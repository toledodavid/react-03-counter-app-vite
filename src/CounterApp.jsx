import PropTypes from 'prop-types';


export const CounterApp = ({value}) => {
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{value}</h2>
    </>
  );
}

CounterApp.defaulProps = {
  value: 0,
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}


