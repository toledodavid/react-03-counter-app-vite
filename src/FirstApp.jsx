// import { Fragment } from "react";
import PropTypes from 'prop-types';

const newMessage = {
  title: 'Hello',
  message: 'David'
};

const greeting = () => (<p>Good morning!</p>);

const getResult = (a, b) => a + b;

export const FirstApp = ({title, subTitle, name}) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h3>{subTitle}</h3>
      <p>{name}</p>

      <code>{JSON.stringify(newMessage)}</code>

      { greeting() }

      <p>Result: { getResult(1, 6) }</p>
    </>
  );

  // return (
  //   <Fragment>
  //     <h1>Hi</h1>
  //     <p>This is a paragraph</p>
  //   </Fragment>
  // );
}

FirstApp.defaultProps = {
  title: 'Not title',
  subTitle: 0,
  name: 'Juan',
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number,
}