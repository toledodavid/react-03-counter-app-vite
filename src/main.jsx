import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
// import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';

import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <FirstApp title='Hello, I am David' subTitle={1234} />
    {/* <CounterApp value={6} /> */}
  </React.StrictMode>
)