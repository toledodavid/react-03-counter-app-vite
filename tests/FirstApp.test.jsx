import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe('Tests for <FirstApp />', () => {

  test('should match with the snapshot', () => {
    const { container } = render(<FirstApp />);
    expect(container).toMatchSnapshot();
  });

  test('should show the title in an h1 element', () => {
    const title = 'Hi, I am David';
    const {container, getByText} = render(<FirstApp title={title}/>);

    const h1 = container.querySelector('h1');

    expect(getByText(title)).toBeTruthy();
    expect(h1.innerHTML).toContain(title);
  });

});