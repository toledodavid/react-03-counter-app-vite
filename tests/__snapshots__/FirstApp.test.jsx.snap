// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <FirstApp /> should match with the snapshot 1`] = `
<div>
  <h1>
    Not title
  </h1>
  <h3>
    0
  </h3>
  <p>
    Juan
  </p>
  <code>
    {"title":"Hello","message":"David"}
  </code>
  <p>
    Good morning!
  </p>
  <p>
    Result: 
    7
  </p>
</div>
`;
